[GTK][WPE] Fix undefined reference to 'shm_open' and 'shm_unlink'

dpino opened the pull request:
#### b0df95241e2ab3a30b0cd72979f7855642fa08a0
<pre>
[GTK][WPE] Fix undefined reference to &apos;shm_open&apos; and &apos;shm_unlink&apos;
<a href="https://bugs.webkit.org/show_bug.cgi?id=268619">https://bugs.webkit.org/show_bug.cgi?id=268619</a>

Reviewed by NOBODY (OOPS!).

Library &apos;rt&apos; is missing in several targets.

* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBufferSHM):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Tools/TestWebKitAPI/PlatformGTK.cmake:
* Tools/TestWebKitAPI/PlatformWPE.cmake:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0df95241e2ab3a30b0cd72979f7855642fa08a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37482 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16366 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39745 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40008 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33404 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38793 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18982 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31810 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38047 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13803 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32889 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12013 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11993 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33573 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41271 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33886 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33994 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37891 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12554 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10127 "Found 2 new test failures: http/tests/webgpu/webgpu/api/operation/render_pipeline/primitive_topology.html, http/tests/webgpu/webgpu/api/operation/rendering/stencil.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36048 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14015 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12961 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13330 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (changes requested):
Needing `librt` for `shm_{open,unlink}` is a GNUism. There is nothing that prevents other C libraries from providing the symbols in the main `libc.so` shared object. In fact both Musl and uCLibc do that, and we support them; OpenBSD also stashes these symbols in `libc.so`, and so on.

What we need to do is check with CMake whether the functions exist without linking any particular library, and if that failed then try whether `librt` provides the symbols.

We have such code in `Source/WebKit/CMakeLists.txt`, around line 664:

```cmake
if (UNIX)
    check_function_exists(shm_open SHM_OPEN_EXISTS)
    if (NOT SHM_OPEN_EXISTS)
        set(CMAKE_REQUIRED_LIBRARIES rt)
        check_function_exists(shm_open SHM_OPEN_REQUIRES_LIBRT)
        if (SHM_OPEN_REQUIRES_LIBRT)
            list(APPEND WebKit_PRIVATE_LIBRARIES rt)
        endif ()
        unset(CMAKE_REQUIRED_LIBRARIES)
    endif ()  
endif ()
```

You would need to investigate why the snippet above is not working as intended. I see the missing symbols were only used in `Source/WebKit/` and the `SharedMemory` classes were moved recently to WebCore... I suspect moving this snippet to `Source/WebCore/CMakeLists.txt` and adjusting so it applies to the `WebCore` target would be the proper fix.

dpino committed:
[GTK][WPE] Fix undefined reference to 'shm_open' and 'shm_unlink'
https://bugs.webkit.org/show_bug.cgi?id=268619

Reviewed by NOBODY (OOPS!).

Library 'rt' is missing in several targets.

* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBufferSHM):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Tools/TestWebKitAPI/PlatformGTK.cmake:
* Tools/TestWebKitAPI/PlatformWPE.cmake:
